zerver/tests/test_notifications.py

```diff
@@ -113,6 +120,82 @@ def _extra_context_in_huddle_missed_stream_messages(self, send_as_user, mock_ran
                 ' the Moor of Venice Group personal message')
         self._test_cases(tokens, msg_id, body, send_as_user)
 
+    @patch('zerver.lib.email_mirror.generate_random_token')
+    def _deleted_message_in_missed_stream_messages(self, send_as_user, mock_random_token):
+        # type: (bool, MagicMock) -> None
+        tokens = self._get_tokens()
+        mock_random_token.side_effect = tokens
+
+        msg_id = self.send_message("othello@zulip.com", "denmark", Recipient.STREAM, '@**hamlet** to be deleted')
+
+        othello = get_user_profile_by_email('othello@zulip.com')
+        (message, user_message) = access_message(othello, msg_id)
+        ums = UserMessage.objects.filter(
+            message=msg_id,
+            flags=~UserMessage.flags.historical)
+        message_users = UserProfile.objects.select_related().filter(
+            id__in={um.user_profile_id for um in ums})
+        content = '(deleted)'
```

Comment:
Instead of manually creating content here, can you send a request to `update_message_backend`? Like this:
```Python
self.client_patch('/messages/<message_id?')
```
and then just do
```Python
self.assertEqual(len(mail.outbox), 0)
```
So your test will just become:
```Python
         tokens = self._get_tokens()                                                                  
         mock_random_token.side_effect = tokens                                                       
         msg_id = self.send_message("othello@zulip.com", "denmark", Recipient.STREAM, '@**hamlet** to be deleted')                     
         self.client_patch()
         self.assertEqual(len(mail.outbox), 0)
```
This way your code will not be dependent on the value '(deleted)'. You also will not have to change `_test_cases` function.